src/data/nat/basic.lean

```diff
@@ -745,11 +745,17 @@ lemma lt_pow_self {p : ℕ} (h : 1 < p) : ∀ n : ℕ, n < p ^ n
   n + 1 < p^n + 1 : nat.add_lt_add_right (lt_pow_self _) _
     ... ≤ p ^ (n+1) : pow_lt_pow_succ h _
 
-lemma le_right_of_pow_le {x m n : ℕ} (ha : 2 ≤ x) (k : x^m ≤ x^n) : m ≤ n :=
-le_of_not_lt (λ r, not_le_of_lt (pow_lt_pow_of_lt_right ha r) k)
+lemma pow_strict_mono {x : ℕ} (k : 2 ≤ x) : strict_mono (nat.pow x) :=
+λ _ _, pow_lt_pow_of_lt_right k
 
-lemma pow_right_inj {x m n : ℕ} (ha : 2 ≤ x) (k : x^m = x^n) : m = n :=
-by apply le_antisymm; apply le_right_of_pow_le ha; apply le_of_eq; rw k
+lemma le_right_iff_pow_le {x m n : ℕ} (k : 2 ≤ x) : x^m ≤ x^n ↔ m ≤ n :=
```

Comment:
Hmm... I now notice that the names seem backwards: `pow_le_iff_le` seems better.